packages/amplify-graphql-model-transformer/src/resources/amplify-dynamodb-table/amplify-table-manager-lambda/import-table.ts

```diff
@@ -0,0 +1,293 @@
+  Pick<
+    TableDescription,
+    | 'AttributeDefinitions'
+    | 'KeySchema'
+    | 'GlobalSecondaryIndexes'
+    | 'BillingModeSummary'
+    | 'ProvisionedThroughput'
+    | 'StreamSpecification'
+    | 'SSEDescription'
+    | 'DeletionProtectionEnabled'
+  >
+>;
+
+/**
+ * Get the expected table properties given the input properties.
+ * This is used to ensure the imported table matches the input properties.
+ *
+ * @param createTableInput input properties for the table creation.
+ */
+export const getExpectedTableProperties = (createTableInput: CreateTableCommandInput): TableComparisonProperties => {
+  return {
+    AttributeDefinitions: getExpectedAttributeDefinitions(createTableInput),
+    KeySchema: getExpectedKeySchema(createTableInput),
+    GlobalSecondaryIndexes: getExpectedGlobalSecondaryIndexes(createTableInput),
+    BillingModeSummary: getExpectedBillingModeSummary(createTableInput),
+    ProvisionedThroughput: getExpectedProvisionedThroughput(createTableInput),
+    StreamSpecification: getExpectedStreamSpecification(createTableInput),
+    SSEDescription: getExpectedSSEDescription(createTableInput),
+    DeletionProtectionEnabled: getExpectedDeletionProtectionEnabled(createTableInput),
+  };
+};
+
+const getExpectedAttributeDefinitions = (createTableInput: CreateTableCommandInput): TableComparisonProperties['AttributeDefinitions'] => {
+  return createTableInput.AttributeDefinitions;
+};
+
+const getExpectedKeySchema = (createTableInput: CreateTableCommandInput): TableComparisonProperties['KeySchema'] => {
+  return createTableInput.KeySchema;
+};
+
+const getExpectedGlobalSecondaryIndexes = (
+  createTableInput: CreateTableCommandInput,
+): TableComparisonProperties['GlobalSecondaryIndexes'] => {
+  return createTableInput.GlobalSecondaryIndexes?.map((gsi) => {
+    if (createTableInput.BillingMode === 'PAY_PER_REQUEST') {
+      // When the billing mode is PAY_PER_REQUEST, the ProvisionedThroughput read and write capacity units will be 0
+      // even if a different number is supplied
+      return {
+        ...gsi,
+        ProvisionedThroughput: gsi.ProvisionedThroughput
+          ? {
+              ReadCapacityUnits: 0,
+              WriteCapacityUnits: 0,
+            }
+          : undefined,
+      };
+    }
+    return gsi;
+  });
+};
+
+const getExpectedBillingModeSummary = (createTableInput: CreateTableCommandInput): TableComparisonProperties['BillingModeSummary'] => {
+  return {
+    BillingMode: createTableInput.BillingMode,
+  };
+};
+
+const getExpectedStreamSpecification = (createTableInput: CreateTableCommandInput): TableComparisonProperties['StreamSpecification'] => {
+  return createTableInput.StreamSpecification;
+};
+
+const getExpectedProvisionedThroughput = (
+  createTableInput: CreateTableCommandInput,
+): TableComparisonProperties['ProvisionedThroughput'] => {
+  return createTableInput.ProvisionedThroughput || { ReadCapacityUnits: 0, WriteCapacityUnits: 0 };
```

Comment:
Any reasoning for the defaulted throughput values here? Do these match the current defaults we use in provisioning?